scripts/src/public/services/orders.ts

```diff
@@ -276,12 +304,15 @@ export async function submitOrder(
 	await order.save();
 	logger.info("order changed to pending", { orderId });
 
-	if (order.type === "earn") {
+	const earnContext = order.contexts.find(context => context.type === "earn");
+	if (earnContext) {
 		await payment.payTo(walletAddress, appId, order.amount, order.id, logger);
-		createEarnTransactionBroadcastToBlockchainSubmitted(order.recipient!.id, order.offerId, order.id).report();
+		createEarnTransactionBroadcastToBlockchainSubmitted(earnContext.user!.id, order.offerId, order.id).report();
```

Comment:
why not recipient

Reply:
the `OrderContext` object has no `recipient`, only `user`